zerver/lib/email_validation.py

```diff
@@ -126,8 +125,16 @@ def get_existing_user_errors(
     already exist.  There's a bit of fiddly logic related
     to cross-realm bots and mirror dummies too.
     '''
+
     errors = {}  # type: Dict[str, Tuple[str, Optional[str], bool]]
 
+    users = get_users_by_delivery_email(emails, target_realm).only(
+        'email',
+        'is_active',
+        'is_mirror_dummy',
+    )
+    user_dict = {user.email: user for user in users}
```

Comment:
I think this logic has a case bug -- if the user submits an email in funny case, it may have a case-insensitive match in `get_users_by_delivery_email` but miss the actual match below.

We should probably just do `.lower()` both here and in a `existing_user_profile = user_dict.get(email.lower())` below.

Reply:
I fixed this and added a test to prevent future similar bugs.  (I pushed the test to the front before fixing the above, and it was an actual failing test till I addressed your comment, so we should be good.)